unit_testing/test_metrics.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+# pytest unit tests for spinalcordtoolbox.process_seg
+
+import numpy as np
+import nibabel as nib
+import csv
+import pytest
+from skimage.transform import rotate
+from spinalcordtoolbox import process_seg
+
+
+@pytest.fixture(scope="session")
+def dummy_segmentation():
+    """Create a dummy image with a ellipse or ones running from top to bottom in the 3rd dimension, and rotate the image
+    to make sure that compute_csa and compute_shape properly estimate the centerline angle."""
+    nx, ny, nz = 20, 20, 20  # image dimension
+    fname_seg = 'dummy_segmentation.nii.gz'  # output seg
+    data = np.random.random((nx, ny, nz))
+    xx, yy = np.mgrid[:nx, :ny]
+    # loop across slices and add an ellipse of axis length a and b
+    a, b = 5.0, 3.0  # diameter of ellipse
+    for iz in range(nz):
+        data[:, :, iz] = (((xx - nx / 2) / a) ** 2 + ((yy - ny / 2) / b) ** 2 <= 1) * 1
+    # swap x-z axes (to make a rotation within y-z plane)
+    data_swap = data.swapaxes(0, 2)
+    # rotate by 15 deg, and re-grid using nearest neighbour interpolation (compute_shape only takes binary iputs)
+    data_swap_rot = rotate(data_swap, 15, resize=False, center=None, order=0, mode='constant', cval=0, clip=True,
+                           preserve_range=False)
+    # swap back
+    data_rot = data_swap_rot.swapaxes(0, 2)
+    xform = np.eye(4)
+    img = nib.nifti1.Nifti1Image(data_rot, xform)
+    nib.save(img, fname_seg)
```

Comment:
return img instead... then APIs can use `Image(img)` or `Image(fname)` to abstract that stuff away, and save on an as-needed basis to temp folders if the path `is None`.